docusaurus/docs/create-a-plugin/extend-a-plugin/nested-plugins.md

```diff
@@ -0,0 +1,87 @@
+---
+id: nested-plugins
+title: Work with nested plugins
+description: How to work with nested plugins, Grafana app plugins.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - advanced
+  - apps
+  - app plugins
+  - nested
+---
+
+import ScaffoldNPM from '@snippets/createplugin-scaffold.npm.md';
+import ScaffoldPNPM from '@snippets/createplugin-scaffold.pnpm.md';
+import ScaffoldYarn from '@snippets/createplugin-scaffold.yarn.md';
+
+Grafana app plugins can nest data sources, both frontend and backend, together with panel plugins so that you can provide a complete user experience.
+
+## Before you begin
+
+See the tutorial to [build an app plugin](../../tutorials/build-an-app-plugin) and scaffold an app plugin.
+
+## Anatomy of nested plugins
+
+Nested plugins are located inside the app plugin `src` folder. They generally follow the same structure of a plugin and have their own `plugin.json`, but they don't have their own `package.json` or `.config` folder.
+
+Here's an example of a nested data source plugin:
+
+```diff bash
+./src
+ ├── README.md
+ ├── components
++├── nested-datasource
++│   ├── components
++│   │   ├── ConfigEditor.tsx
++│   │   └── QueryEditor.tsx
++│   ├── datasource.ts
++│   ├── img
++│   ├── module.ts
++│   ├── plugin.json
++│   └── types.ts
+ ├── img
+ │   └── logo.svg
+ ├── module.ts
+ └── plugin.json
+```
+
+:::note
+
+Notice that the plugin `nested-datasource` doesn't have a `package.json` of its own. The name of the nested plugin folder is not important.
+
+:::
+
+## How to add a nested plugin to an app plugin
+
+1. Generate the nested plugin:
+
+   :::important
+
+   Begin outside your app plugin's directory.
+
+   :::
+
+   Use the `create-plugin` tool to generate a new plugin:
+
+   ```bash
+   npx @grafana/create-plugin@latest
+   ```
+
+   Carefully select the desired plugin type (data source or panel), provide a name, and use the same organization as your app plugin.
+
+1. Prepare the nested plugin:
+
+   Rename the `src` folder of your newly generated plugin to a name that reflects its specific purpose (for example, `nested-datasource`).
+
+1. Integrate into your app plugin:
```

Comment:
Should reference what is possible with backend data sources / backend apps

Reply:
@sympatheticmoose I am not sure what you mean here. can you suggest something to add?

Reply:
For example - if you want your app to nest a backend datasource, you would likely create the backend datasource separately for the scaffolded code and then need to copy more than just `src`. For example the zabbix plugin https://github.com/grafana/grafana-zabbix where the app doesn't claim to have a backend, but the nested DS does

Reply:
The exact details I would speak to the team about what is and isnt possible, I just feel theres more missing here

Reply:
I added a section on top of this one. I don't think this specific step is the best place to explain about backend datasources.